skstan/base.py

```diff
@@ -1,13 +1,12 @@
 from abc import ABCMeta, abstractstaticmethod
-from collections import UserDict
+from typing import Dict
 
 
-class BaseStanData(UserDict):
+class BaseStanData(Dict):
     def append(self, **kwargs):
-        self.data.update(kwargs)
-        return BaseStanData(
-            self.data
-        )
+        data = self.copy()
+        data.update(kwargs)
+        return data
```

Comment:
I think new data structures which we define in this project should be put in one place independently. for example, under `utils/collection/...`.
And, that immutability should be explicitly declared for improving code readability.
```python
from typing import Dict

class ImmutableDict(Dict):
    def append(self, **kwargs):
        ....
```

```python
class BaseStanData(ImmutableDict):
    ...
```